airbyte-integrations/connectors/source-airtable/source_airtable/manifest.yaml

```diff
@@ -0,0 +1,142 @@
+      name: custom_object
+      $parameters:
+        table_id: "table_id"
+      retriever:
+        requester:
+          $ref: "#/definitions/base_requester"
+          path: /{{stream_partition['parent_slice']['id']}}/{{parameters['table_id']}}
+          http_method: GET
+        partition_router:
+          type: SubstreamPartitionRouter
+          parent_stream_configs:
+            - type: ParentStreamConfig
+              parent_key: id
+              partition_field: id
+              stream:
+                $ref: "#/definitions/streams/tables"
+        record_selector:
+          extractor:
+            type: DpathExtractor
+            field_path: ["records"]
+      schema_loader:
+        type: DynamicSchemaLoader
+        stream:
+          type: DeclarativeStream
+          name: tables
+          $parameters:
+            table_id: "table_id"
+          retriever:
+            requester:
+              $ref: "#/definitions/base_requester"
+              path: meta/bases/{{stream_partition['id']}}/tables
+              http_method: GET
+            partition_router:
+              type: SubstreamPartitionRouter
+              parent_stream_configs:
+                - type: ParentStreamConfig
+                  parent_key: id
+                  partition_field: id
+                  stream:
+                    $ref: "#/definitions/streams/bases"
+            record_selector:
+              extractor:
+                type: DpathExtractor
+                field_path: ["tables"]
+              record_filter:
+                condition: "{{ record['id'] == parameters['table_id'] }}"
+        schema_pointer: ["fields"]
+        key_pointer: ["name"]
+        type_pointer: ["type"]
+        types_map:
+          - target_type: string
+            current_type: singleLineText
+          - target_type: string
+            current_type: multilineText
+  base_requester:
+    type: HttpRequester
+    url_base: https://api.airtable.com/v0
+    authenticator: "#/definitions/authenticator"
+  # Authenticators
+  bearer_authenticator:
+    type: BearerAuthenticator
+    api_token: "{{ config['credentials']['api_key'] }}"
+  oauth_authenticator:
+    type: OAuthAuthenticator
+    refresh_request_body: {}
+    token_refresh_endpoint: https://airtable.com/oauth2/v1/token
+    grant_type: refresh_token
+    client_id: '{{ config["credentials"]["client_id"] }}'
+    client_secret: '{{ config["credentials"]["client_secret"] }}'
+    refresh_token: '{{ config["credentials"]["refresh_token"] }}'
+  authenticator:
+    type: SelectiveAuthenticator
+    authenticator_selection_path: ["credentials", "auth_method"]
+    authenticators:
+      oauth2: "#/definitions/oauth_authenticator"
+      api_key: "#/definitions/bearer_authenticator"
+
+streams:
+  - $ref: "#/definitions/streams/custom_object"
+
+dynamic_streams:
+  - type: DynamicDeclarativeStream
+    stream_template:
+      $ref: "#/definitions/streams/custom_object"
+    components_parser:
```

Comment:
Can we check how this could work given [source-salesforce case where there are conditions for us to consider](https://github.com/airbytehq/airbyte/blob/7ac310010e88ae677aa1218624fc5038bb7ada3b/airbyte-integrations/connectors/source-salesforce/source_salesforce/api.py#L49-L219)? I assume that the filtering can be done as a record filter on the table. I am mostly worried about logic like "if name == ContentDocumentLink then parent stream else normal stream"

Reply:
I have added some example of this to the implementation. This can be achieved with a few conditional checks in the record extractor.

Reply:
I don't see the change in source-salesforce as I don't see any `SubstreamPartitionRouter`. So can you confirm my understanding: we would have multiple `dynamic_streams` for source-salesforce and the components_parser.retriever would filter on the entries that are specific for each type of stream. Is that correct?